src/localconf.js

```diff
@@ -160,6 +162,8 @@ function buildConfig(opts) {
       ddSampleRate = config.ddSampleRate;
     }
   }
+  const ddKnownBotPattern = crawlers.map((c) => c.pattern).join('|');
+  const ddKnownBotRegex = new RegExp(ddKnownBotPattern, 'i');
```

Comment:
So this takes all the patterns from the crawler-user-agents.json here: https://github.com/monperrus/crawler-user-agents/blob/master/crawler-user-agents.json
 
and creates a regex ORing them all together to give us our ddKnownBotRegex

Then our conditional sample rate test if the user agent from a "user" matches on that regex - if it does then we don't count the sample otherwise we do? 

Is my understanding on that correct?

Reply:
yeah you are right, if the value of user agent matches the pattern of known crawlers, then we set the DD RUM session sample rate to 0, which mean no sessions are being send to DD

this method is documented in the official DD RUM user doc https://docs.datadoghq.com/real_user_monitoring/guide/identify-bots-in-the-ui/?s=userAgent#filter-out-bot-sessions-on-intake